components/modal/Modal.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useCallback, useEffect, useRef } from 'react';
+import {
+  ButtonIconRounded,
+  EIConTypes,
+} from '@ost-cas-fee-adv-23-24/elbmum-design';
+import useModal from '@/hooks/useModal';
+
+const escapeKeyCode = 27;
+const Modal = () => {
+  const { content, isOpen, closeModal, title, fullWidth } = useModal();
+  const positionCommon = 'top-0 right-0 bottom-0 left-0';
+  const transitionOptions = 'opacity duration-300 ease delay-300';
+
+  const escapeKeyUpHandler = useCallback(
```

Comment:
is the useCallback necessary? in the design system review the teachers weren't happy with our use of useCallback usage

Reply:
In this case I would keep using it. The function would be attached on every single rendering of the component. We have only one instance of the modal, which is ready to get content. 
We can also ask  Mirko, if you are unsure.